test/search/test_octree.cpp

```diff
@@ -301,51 +301,43 @@ TEST (PCL, Octree_Pointcloud_Neighbours_Within_Radius_Search)
                                      static_cast<float> (5.0 *  (rand () / static_cast<double> (RAND_MAX))));
     }
 
-    pcl::search::Search<PointXYZ>* octree = new pcl::search::Octree<PointXYZ> (0.001);
+    pcl::search::Octree<PointXYZ> octree(0.001);
 
     // build octree
-    double pointDist;
     double searchRadius = 5.0 * rand () / static_cast<double> (RAND_MAX);
     
     // bruteforce radius search
-    std::vector<int> cloudSearchBruteforce;
+    std::size_t cloudSearchBruteforce_size_lower = 0, cloudSearchBruteforce_size_upper = 0;
     for (std::size_t i = 0; i < cloudIn->size (); i++)
     {
-      pointDist = std::sqrt (
-                        ((*cloudIn)[i].x - searchPoint.x) * ((*cloudIn)[i].x - searchPoint.x)
-                            + ((*cloudIn)[i].y - searchPoint.y) * ((*cloudIn)[i].y - searchPoint.y)
-                            + ((*cloudIn)[i].z - searchPoint.z) * ((*cloudIn)[i].z - searchPoint.z));
-
-      if (pointDist <= searchRadius)
-      {
-        // add point candidates to vector list
-        cloudSearchBruteforce.push_back (static_cast<int> (i));
+      const auto pointDist = pcl_tests::point_distance((*cloudIn)[i], searchPoint);
+      if (pointDist <= (searchRadius+TOLERANCE)) {
+        ++cloudSearchBruteforce_size_upper;
+        if (pointDist <= (searchRadius-TOLERANCE)) {
+          ++cloudSearchBruteforce_size_lower;
+        }
       }
     }
 
     pcl::Indices cloudNWRSearch;
     std::vector<float> cloudNWRRadius;
 
     // execute octree radius search
-    octree->setInputCloud (cloudIn);
-    octree->radiusSearch (searchPoint, searchRadius, cloudNWRSearch, cloudNWRRadius);
+    octree.setInputCloud (cloudIn);
+    octree.radiusSearch (searchPoint, searchRadius, cloudNWRSearch, cloudNWRRadius);
 
-    ASSERT_EQ ( cloudNWRRadius.size() , cloudSearchBruteforce.size());
+    ASSERT_GE ( cloudNWRRadius.size() , cloudSearchBruteforce_size_lower);
+    ASSERT_LE ( cloudNWRRadius.size() , cloudSearchBruteforce_size_upper);
 
-    // check if result from octree radius search can be also found in bruteforce search
+    // check if results from organized radius search are indeed within the search radius
```

Comment:
Aren't this still results from an octree search?

Reply:
Thanks, you are right. It was a copy-and-paste error